Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -3530,6 +3530,13 @@ public NameSegment(IToken tok, string name, List<Type> optTypeArguments)
   }
 
   public override IEnumerable<Node> PreResolveChildren => OptTypeArguments ?? new List<Type>();
+  public bool SetIndent(int indentBefore, TokenNewIndentCollector formatter) {
+    formatter.SetTypeLikeIndentation(indentBefore, OwnedTokens);
+    foreach (var subType in PreResolveChildren.OfType<Type>()) {
+      formatter.SetTypeIndentation(subType);
+    }
+    return false;
+  }
```

Comment:
Without this line, the tests containing
```
  ConstructorName(
    a: LongName < map <
           b,
           c>>
```
Were incorrectly formatted post-resolution, because the user defined type becomes a NameSegment expression after resolution